scripts/submit_bb.py

```diff
@@ -36,129 +37,178 @@ def gen_command_template(params):
-
-        wct = set_wct(est_run_time_scaled, ncores, args.auto)
-        write_directory = (
-            args.write_directory if args.write_directory else params.sim_dir
-        )
-        underscored_srf = srf_name.replace("/", "__")
-
-        header_dict = {
-            "wallclock_limit": wct,
-            "job_name": "bb.{}".format(underscored_srf),
-            "job_description": "BB calculation",
-            "additional_lines": "###SBATCH -C avx",
-            "platform_specific_args": get_platform_node_requirements(ncores),
-        }
-
-        body_template_params = (
-            "{}.sl.template".format(sl_name_prefix),
-            {"test_bb_script": "test_bb.sh"},
-        )
+    srf_name = Path(params.srf_file).stem
+
+    # TODO: save status as HF. refer to submit_hf
+    # Use HF nt for wct estimation
+    nt = get_hf_nt(params)
+    fd_count = len(shared.get_stations(params.FD_STATLIST))
+
+    est_core_hours, est_run_time = est.est_BB_chours_single(fd_count, nt, ncores)
+
+    # creates and extra variable so we keep the original estimated run time for other purpos
+    est_run_time_scaled = est_run_time
+    if retries > 0:
+        # check if BB.bin is read-able = restart-able
+        try:
+            from qcore.timeseries import BBSeis
+
+            bin = BBSeis(simulation_structure.get_bb_bin_path(params.sim_dir))
+        except:
+            logger.debug("Retried count > 0 but BB.bin is not readable")
+        else:
+            est_run_time_scaled = est_run_time * (retries + 1)
+
+    wct = set_wct(est_run_time_scaled, ncores, auto)
+    write_directory = write_directory if write_directory else params.sim_dir
+    if write_directory is None:
+        write_directory = params.sim_dir
+
+    underscored_srf = srf_name.replace("/", "__")
+
+    header_dict = {
+        "wallclock_limit": wct,
+        "job_name": f"bb.{underscored_srf}",
+        "job_description": "BB calculation",
+        "additional_lines": "###SBATCH -C avx",
+        "platform_specific_args": get_platform_node_requirements(ncores),
+    }
+
+    body_template_params = (
+        "{}.sl.template".format(sl_name_prefix),
+        {"test_bb_script": "test_bb.sh"},
+    )
 
-        command_template_parameters, add_args = gen_command_template(params)
+    command_template_parameters, add_args = gen_command_template(params)
+
+    script_prefix = f"{sl_name_prefix}_{underscored_srf}"
+    script_file_path = write_sl_script(
+        write_directory,
+        params.sim_dir,
+        const.ProcessType.BB,
+        script_prefix,
+        header_dict,
+        body_template_params,
+        command_template_parameters,
+        add_args,
+    )
 
-        script_prefix = "{}_{}".format(sl_name_prefix, underscored_srf)
-        script_file_path = write_sl_script(
-            write_directory,
+    # Submit the script
+    submit_yes = True if auto else confirm("Also submit the job for you?")
```

Comment:
Do we need to use confirm here? We can use the argument to specify submit if we want to, otherwise just create the slurm/pbs file

Reply:
--submit argument is by default False, and it will take care of it.